### ISSUE TITLE
MD041 and github templates

### ISSUE THREAD

**Issue by hepcat72** [DavidAnson/markdownlint#380]
I have a question.  I created github templates for issues and pull requests and I'm confused about the default behavior of MD041.

The following github-recognized default-generated templates for github issues do not trigger an MD041 error:

```
---
name: Bug Report
about: Create a report to help us improve
title: ''
labels: bug
assignees: ''

---

## BUG DESCRIPTION

### Problem

A bird's-eye summation of what the problem is.

...
```

Github handles the `---`-delimited portion so that issue submitters do not see it.  Their default pull request template however, does not include this `---`-delimited section, and if you add one (there is no MD041 error *but*), submitters will see this section.  If I remove this section, the MD041 error occurs.

I feel that an `h1` is too big for the PR template.  Is there a work-around so that other docs require an `h1`, but the PR template doesn't have to have one?

**Comment by DavidAnson** [comment]
My suggestion might be to use an inline comment to customize the exceptional file: https://github.com/DavidAnson/markdownlint#configuration

Something like this at the bottom of the template should do what you want without getting in the way: `<!-- markdownlint-disable-file MD041 -->`

**Comment by QuAzI** [comment]
MD041 looks wrong because all note software (QOwnNotes, Obsidian, Foam...) use `---` to split title (HEAD) of document and h1...5 just for sectioning BODY
```
Some title
---

File content
```

**Comment by DavidAnson** [comment]
That syntax creates a second-level/H2 heading. Rule MD041 advocates for a top-level/H1 heading as the first line of the document. As such, I believe it is correct to report your example as a violation.

https://spec.commonmark.org/0.30/#setext-headings

https://cirosantilli.com/markdown-style-guide/#top-level-header

http://markdown-it.github.io/#md3=%7B%22source%22%3A%22Some%20title%5Cn---%5Cn%5CnFile%20content%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22src%22%7D%7D
